statistics/scalar.go

```diff
@@ -175,3 +177,122 @@ func calcFraction4Datums(lower, upper, value *types.Datum) float64 {
 	}
 	return 0.5
 }
+
+const maxNumStep = 10
```

Comment:
Could we make this a variable correlated with average bucket width for example?

Reply:
But we still need a `max` step to limit the optimization time.